core/src/main/java/com/yahoo/oak/Block.java

```diff
@@ -37,12 +37,21 @@ void setID(int id) {
     // Block manages its linear allocation. Thread safe.
     // The returned buffer doesn't have all zero bytes.
     boolean allocate(Slice s, int size) {
-        int now = allocated.getAndAdd(size);
-        if (now + size > this.capacity) {
-            allocated.getAndAdd(-size);
+        assert size >= 0; // avoid negative size
```

Comment:
We probably want to avoid `size==0` as well.

Reply:
@liran-funaro Sure, will update the check as `assert size > 0`. I initially added `size >= 0`(included 0) to be backward compatible with any existing scenario using `0` size. 
Logically, users of this method should not pass `0` too, so will update check.
👍